prompt/prompt-executor/prompt-executor-clients/prompt-executor-bedrock-client/src/jvmTest/kotlin/ai/koog/prompt/executor/clients/bedrock/BedrockLLMClientTest.kt

```diff
@@ -332,6 +339,129 @@ class BedrockLLMClientTest {
+
+        val client = BedrockLLMClient(
+            mockClient,
+            moderationGuardrailsSettings = guardrailsSettings
+        )
+
+        try {
+            // Prompt with only User (Request) message
+            val prompt = Prompt.build("test") {
+                user("hi")
+            }
+            val model = BedrockModels.AnthropicClaude3Sonnet
+
+            client.moderate(prompt, model)
+
+            assertEquals(1, applyGuardrailCallCount, "Should call applyGuardrail exactly once for Request-only prompts")
+        } finally {
+            client.close()
+        }
+    }
+
+    @Execution(ExecutionMode.SAME_THREAD)
+    @Test
+    fun `moderate calls guardrails twice for prompts with both Request and Response`() = runTest {
+        val guardrailsSettings = BedrockGuardrailsSettings(
+            guardrailIdentifier = "test-guardrail",
+            guardrailVersion = "1.0"
+        )
+
+        var applyGuardrailCallCount = 0
+
+        val mockClient = createCountingMockClient { applyGuardrailCallCount++ }
+
+        val client = BedrockLLMClient(
+            mockClient,
+            moderationGuardrailsSettings = guardrailsSettings
+        )
+
+        try {
+            // Prompt with both User (Request) and Assistant (Response) messages
+            val prompt = Prompt.build("test") {
+                user("What is 2+2?")
+                assistant("2+2 equals 4")
+            }
+            val model = BedrockModels.AnthropicClaude3Sonnet
+
+            client.moderate(prompt, model)
+
+            assertEquals(2, applyGuardrailCallCount, "Should call applyGuardrail exactly twice for prompts with both Request and Response")
+        } finally {
+            client.close()
+        }
+    }
+
+    @Execution(ExecutionMode.SAME_THREAD)
+    @Test
+    fun `moderate calls guardrails once for Response-only prompts`() = runTest {
+        val guardrailsSettings = BedrockGuardrailsSettings(
+            guardrailIdentifier = "test-guardrail",
+            guardrailVersion = "1.0"
+        )
+
+        var applyGuardrailCallCount = 0
+
+        val mockClient = createCountingMockClient { applyGuardrailCallCount++ }
+
+        val client = BedrockLLMClient(
+            mockClient,
+            moderationGuardrailsSettings = guardrailsSettings
+        )
+
+        try {
+            // Prompt with only Assistant (Response) message
+            val prompt = Prompt.build("test") {
+                assistant("Hello, how can I help?")
+            }
+            val model = BedrockModels.AnthropicClaude3Sonnet
+
+            client.moderate(prompt, model)
+
+            assertEquals(1, applyGuardrailCallCount, "Should call applyGuardrail exactly once for Response-only prompts")
+        } finally {
+            client.close()
+        }
+    }
+
+    // Helper function to create a counting mock client
+    private fun createCountingMockClient(onApplyGuardrail: () -> Unit): BedrockRuntimeClient {
```

Comment:
nit: leaving `TODO` in the actual code produces unnecessary warnings, let's throw more descriptive `UnsupportedOperationException`